snuba/replacer.py

```diff
@@ -332,12 +349,26 @@ def run_query(
         )
 
     def process_message(self, message: Message[KafkaPayload]) -> Optional[Replacement]:
+        metadata = ReplacementMessageMetadata(
+            partition_index=message.partition.index, offset=message.offset,
+        )
+        if self._message_already_processed(metadata):
+            logger.warning(
+                f"Replacer ignored a message, consumer group: {self.__consumer_group}",
+                extra={
+                    "partition": metadata.partition_index,
+                    "offset": metadata.offset,
+                },
+            )
+            return None
```

Comment:
You should rollout this behind a feature flag just in case.
Add a runtime flag here, that, when you rollout, will be disabled, if the flag is disabled jsut log and go on. If the flag is enabled, actually return None.
This allows you to do a safe deployment in case of bugs. Then you can activate and remove the flag once you are sure you are not logging these messages.